### PR TITLE
Add Test for components

### PR DESCRIPTION
**Pull request by dinkrr**
Added tests using react testing library for all the compoenents

### CODE REVIEW

File: src/components/counter/__tests__/Counter.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import Counter from '../counter';
+
+test('renders the counter with initial value', () => {
```
Comment (AndreiyMarkusheuski) [inline src/components/counter/__tests__/Counter.test.js:5]: Will be better to cover all those tests into one common (by component):

```
describe('Counter should work as expected',() => {
     it('renders the counter with initial value', () => {...});
     it('increments the count when the "+" button is clicked', () => {...});
})
```

Not required, but will easy to read in future

File: src/components/counter/__tests__/Counter.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const { getByText } = render(<Counter />);
+  const countElement = getByText('0');
+  expect(countElement).toBeInTheDocument();
+});
+
+test('increments the count when the "+" button is clicked', () => {
+  const { getByText } = render(<Counter />);
+  const incrementButton = getByText('+');
+  const countElement = getByText('0');
```
Comment (AndreiyMarkusheuski) [inline src/components/counter/__tests__/Counter.test.js:14]: Search by text isn't a good practice. 
Instead of it, use `querySelector` or `getElementById`

**Review by AndreiyMarkusheuski — APPROVED** [review]

File: src/components/counter/__tests__/Counter.test.js (hunk 1 above)
Comment (dinkrr) [reply src/components/counter/__tests__/Counter.test.js]: done.
thanks

File: src/components/counter/__tests__/Counter.test.js (hunk 2 above)
Comment (dinkrr) [reply src/components/counter/__tests__/Counter.test.js]: Thanks for suggestion. Implemented query-selector.

**Comment by dinkrr** [comment]
Added an e2e test using cypress.

**Pull request merged**